railties/lib/rails/all.rb

```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+# rubocop:disable Style/RedundantBegin
+
```

Comment:
I remember that [we once decided  not to spread any rubocop's comments in the code base](https://github.com/rails/rails/pull/32381#discussion_r205212331), and [exclude cops for specific file/s via `.rubocop.yml` file](https://github.com/rails/rails/pull/33441) if needed:
```diff
diff --git a/.rubocop.yml b/.rubocop.yml
index 96e2ad76d0..3c85378dc0 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -189,6 +189,8 @@ Style/ParenthesesAroundCondition:

 Style/RedundantBegin:
   Enabled: true
+  Exclude:
+    - 'railties/lib/rails/all.rb'

 Style/RedundantReturn:
   Enabled: true
diff --git a/railties/lib/rails/all.rb b/railties/lib/rails/all.rb
index 7f4c625b39..f5dccd2381 100644
--- a/railties/lib/rails/all.rb
+++ b/railties/lib/rails/all.rb
@@ -1,7 +1,5 @@
 # frozen_string_literal: true

-# rubocop:disable Style/RedundantBegin
-
 require "rails"

 %w(
```

Should we follow that approach in this case?

Reply:
IMO no. Configured exclusions make sense when it's structural ("this rule doesn't apply to tests"), whereas comments work better when it's contextual ("this particular instance is an exception to the rule"). Putting the latter in the config makes the config noisy, plus it moves the definition of the exception away from the context that makes it exceptional.

We definitely don't want such comment-exceptions in places where we just feel the rule is making a bad choice: that implies we don't agree with the rule, so warrants deeper consideration... "this file ignores the rule to remain syntactically compatible with older Rubies" is enough of a special case to me.

(If the path to the version check had required more than one instance of the comment, then things might've balanced out differently.)